webapp/app/templates/lotse/form_steuernummer.html

```diff
@@ -0,0 +1,184 @@
+            var selectedState = stateDropdown.options[stateDropdown.selectedIndex].value.toLowerCase();
+            var selectedTaxOffices = taxOffices.find(taxOffice => taxOffice.state_abbreviation === selectedState)
+            let oldBufaNrValue = bufaNrDropdown.value
+            deleteSelectOptions(bufaNrDropdown, defaultOptionText="{{ _('form.select_input.default_selection') }}");
+
+            if(selectedTaxOffices){
+                selectedTaxOffices = selectedTaxOffices.tax_offices;
+                selectedTaxOffices && selectedTaxOffices.sort(function (a, b) {
+                    return ('' + a.name).localeCompare(b.name);
+                });
+                for (var i = 0; i < selectedTaxOffices.length; i++) {
+                  var tax_office = new Option(selectedTaxOffices[i].name.replace("Finanzamt ", ""), selectedTaxOffices[i].bufa_nr);
+                  bufaNrDropdown.options.add(tax_office);
+                }
+                bufaNrDropdown.value = oldBufaNrValue
+            }
+        }
+
+        function changeTaxNumberExists(){
+            let taxNumberExistsYesField = document.querySelector('input[type=radio][name=steuernummer_exists][value="yes"]');
+            let taxNumberExistsNoField = document.querySelector('input[type=radio][name=steuernummer_exists][value="no"]');
+            if (taxNumberExistsYesField.checked && !taxNumberExistsYesField.wasCheckedBefore){
+                console.log("dudu")
+                taxNumberFormState = TAX_NUMBER_FORM_STATES.StateSelectionTaxNumberExists;
+                taxNumberExistsYesField.wasCheckedBefore = true;
+                taxNumberExistsNoField.wasCheckedBefore = false;
+            }else if(taxNumberExistsNoField.checked && !taxNumberExistsNoField.wasCheckedBefore){
+                taxNumberFormState = TAX_NUMBER_FORM_STATES.StateSelectionNoTaxNumberExists;
+                taxNumberExistsYesField.wasCheckedBefore = false;
+                taxNumberExistsNoField.wasCheckedBefore = true;
+            }
+        }
+
+        function changeBundesland(){
+            let bundeslandField = document.getElementById('bundesland');
+            if (bundeslandField.value && taxNumberFormState === TAX_NUMBER_FORM_STATES.StateSelectionTaxNumberExists){
+                taxNumberFormState = TAX_NUMBER_FORM_STATES.TaxNumberInput;
+            }else if(bundeslandField.value
+                     && (taxNumberFormState === TAX_NUMBER_FORM_STATES.StateSelectionNoTaxNumberExists
+                            || taxNumberFormState === TAX_NUMBER_FORM_STATES.TaxOfficeSelection)){
+                taxNumberFormState = TAX_NUMBER_FORM_STATES.TaxOfficeSelection;
+                updateTaxOfficeList();
+            }else if(bundeslandField.value && taxNumberFormState === TAX_NUMBER_FORM_STATES.RequestNewTaxNumber){
+                taxNumberFormState = TAX_NUMBER_FORM_STATES.TaxOfficeSelection;
+                updateTaxOfficeList();
```

Comment:
Why aren't thos two cases the same?

Reply:
You are totally correct. They should be the same.